tests/vm_test.go

```diff
@@ -1969,12 +1969,25 @@ status:
 
 			By("Blocking virt-handler from reconciling the VMI")
 			libpod.AddKubernetesApiBlackhole(getHandlerNodePod(virtClient, nodeName), componentName)
-			Eventually(getHandlerNodePod(virtClient, nodeName).Items[0], 120*time.Second, time.Second, HaveConditionFalse(k8sv1.PodReady))
```

Comment:
How about?
```
Eventually(func(g Gomega) {
	g.Expect(getHandlerNodePod(virtClient, nodeName).Items[0]).To(HaveConditionFalse(k8sv1.PodReady))
}, 120*time.Second, time.Second).Should(Succeed())

DeferCleanup(func() {
	libpod.DeleteKubernetesApiBlackhole(getHandlerNodePod(virtClient, nodeName), componentName)
	Eventually(func(g Gomega) {
		g.Expect(getHandlerNodePod(virtClient, nodeName).Items[0]).To(HaveConditionTrue(k8sv1.PodReady))
	}, 120*time.Second, time.Second).Should(Succeed())
})
```

Reply:
Why did we revert the `defer` or `JustAfterEach`?

Reply:
I don't get it.
The `defer` is something that we can avoid as solution to cleanup. The `JustAfterEach` is generally used for different scopes.
Here, the problem was the `DeferCleanup`. After some searches I found this https://github.com/onsi/ginkgo/issues/1284#issuecomment-1756314394
```
In reality Ginkgo runs the After* family followed by the Defer* family.
```
which explain the source of the issue.
(We should check our other `DeferCleanup` usages IMHO).
Can you explain your doubts?
Thanks

Reply:
@jcanocan also observed that an `AfterEach` is only run after the global `AfterEach` afaik.

Reply:
Thanks @fossedihelm for this nice piece of information. This explains all the issues faced.

As far as I was able to test, the nearest `AfterEach` (the one inside the `Context`) is always executed before the global `AfterEach`. However, we need a way to wait until the `handler` and `kubevirt` versions are in sync. That's why I've observed in the past the error mentioned in the description. Sorry for the confusion.